monitoring/testonly/delta.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testonly
+
+import (
+	"github.com/google/trillian/monitoring"
+)
+
+// CounterSnapshot records the values of a counter at a specific point in time.
+type CounterSnapshot struct {
+	c      monitoring.Counter
+	values map[string]float64
+}
+
+// SnapshotCounter records the current values of a counter.
+// It will only record values associated with the given labels.
+func SnapshotCounter(c monitoring.Counter, labels ...string) CounterSnapshot {
+	s := CounterSnapshot{
+		c:      c,
+		values: make(map[string]float64, len(labels)),
+	}
+	for _, l := range labels {
+		s.values[l] = c.Value(l)
+	}
+	return s
+}
+
+// Delta returns the difference between the current value of a counter and its
+// value when the snapshot was created.
+func (s CounterSnapshot) Delta(label string) float64 {
+	return s.c.Value(label) - s.values[label]
```

Comment:
`Value` accepts a list of labels in a general case. It is a coincidence that Trillian has only 1 label (`treeID`).
Make this function accept `label ...string` instead, in addition to the modification above?

Reply:
Done.